meta-client/src/main/java/org/apache/helix/metaclient/impl/zk/ZkMetaClient.java

```diff
@@ -414,4 +414,130 @@ private static EntryMode convertZkEntryMode(long ephemeralOwner) {
         throw new IllegalArgumentException(zkEphemeralType + " is not supported.");
     }
   }
+
+
+  @Override
+  public List<OpResult> transactionOP(Iterable<Op> iterable) throws KeeperException {
+    // Convert list of MetaClient Ops to Zk Ops
+    List<org.apache.zookeeper.Op> zkOps = metaClientOpToZk(iterable);
+    // Execute Zk transactional support
+    List<org.apache.zookeeper.OpResult> zkResult = _zkClient.multi(zkOps);
+    // Convert list of Zk OpResults to MetaClient OpResults
+    return zkOpResultToMetaClient(zkResult);
+  }
+
+  /**
+   * Helper function for transactionOp. Converts MetaClient Op's into Zk Ops to execute
+   * zk transactional support.
+   * @param ops
+   * @return
+   */
+  public List<org.apache.zookeeper.Op> metaClientOpToZk(Iterable<Op> ops) throws KeeperException {
+    List<org.apache.zookeeper.Op> zkOps = new ArrayList<>();
+    org.apache.zookeeper.Op temp;
+    for (Op op : ops) {
+      switch (op.getType()) {
+        case CREATE:
+          int zkFlag = zkFlagFromEntryMode(((Op.Create) op).getEntryMode());
+          temp = org.apache.zookeeper.Op.create(
+              op.getPath(), ((Op.Create) op).getData(), DEFAULT_ACL, CreateMode.fromFlag(zkFlag));
+          break;
+        case DELETE:
+          temp = org.apache.zookeeper.Op.delete(
+              op.getPath(), ((Op.Delete) op).getVersion());
+          break;
+        case SET:
+          temp = org.apache.zookeeper.Op.setData(
+              op.getPath(), ((Op.Set) op).getData(), ((Op.Set) op).getVersion());
+          break;
+        case CHECK:
+          temp = org.apache.zookeeper.Op.check(
+              op.getPath(), ((Op.Check) op).getVersion());
+          break;
+        default:
+          throw new IllegalArgumentException(op.getType() + " is not supported.");
+      }
+      zkOps.add(temp);
+    }
+    return zkOps;
+  }
+
+  /**
+   * Helper function for transactionOP. Converts the result from calling zk transactional support into
+   * metaclient OpResults.
+   * @param zkResult
+   * @return
+   */
+  public List<OpResult> zkOpResultToMetaClient(List<org.apache.zookeeper.OpResult> zkResult) throws KeeperException.BadArgumentsException {
```

Comment:
ditto.

Reply:
Please define metaclient exception under org.apache.helix.metaclient.constants.

Reply:
Sounds good, will create a specific MetaClientKeeperException for these cases (where KeeperExceptions are thrown)